### PR TITLE
infoblox inventory script: Use stderr, and allow use of environment for config

### PR DESCRIPTION
**Pull request by lingfish**
Originally from https://github.com/ansible/ansible/pull/49685

##### SUMMARY
Errors should send to stderr, and this allows the use of an environment var to point to where the YAML config lives.

This is particularly useful when using this script with AWX and custom credentials (injecting YAML config).

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
infoblox.py

### CODE REVIEW

**Comment by bmillemathias** [comment]
Hi @lingfish 
We required for each pull request to have a changelog file (as [here](https://github.com/ansible-collections/community.general/tree/main/changelogs/fragments)), could you amend that ?

Thanks

File: changelogs/fragments/436-infoblox-use-stderr-and-environment-for-config.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - infoblox.py - infoblox inventory script: Use stderr, and allow use of environment for config (https://github.com/ansible-collections/community.general/pull/436)
```
Comment (felixfontein) [inline changelogs/fragments/436-infoblox-use-stderr-and-environment-for-config.yaml:2]:
```suggestion
  - infoblox inventory script - use stderr for reporting errors, and allow use of environment for configuration (https://github.com/ansible-collections/community.general/pull/436).
```

File: scripts/inventory/infoblox.py [hunk 2]
```diff
@@ -54,7 +55,7 @@ def main():
         if os.path.exists(config_file):
```
Comment (felixfontein) [inline scripts/inventory/infoblox.py:55]:
```suggestion
        if os.path.exists(config_file):
```
I think `os.path.exists` crashes when being passed `None`.

File: scripts/inventory/infoblox.py [hunk 3]
```diff
@@ -30,6 +30,7 @@
 
 
 CONFIG_FILES = [
+    os.environ.get('INFOBLOX_CONFIG_FILE', None),
```
Comment (felixfontein) [inline scripts/inventory/infoblox.py:33]: To avoid the crash:
```suggestion
    os.environ.get('INFOBLOX_CONFIG_FILE', ''),
```

**Comment by Andersson007** [comment]
any updates on this?

File: scripts/inventory/infoblox.py (hunk 3 above)
Comment (lingfish) [reply scripts/inventory/infoblox.py:33]: Agreed.

**Review by felixfontein — APPROVED** [review]
I'm not using the script, but I think the change looks ok. If nobody complains, I'll merge this in a few days.

**Review by Andersson007 — APPROVED** [review]

**Comment by felixfontein** [comment]
@lingfish thanks for improving this script!
@Andersson007 thanks for reviewing!

**Pull request merged**